src/Query/UdpTransportExecutor.php

```diff
@@ -95,11 +96,7 @@ final class UdpTransportExecutor implements ExecutorInterface
      */
     private $maxPacketSize = 512;
 
-    /**
-     * @param string         $nameserver
-     * @param ?LoopInterface $loop
-     */
-    public function __construct($nameserver, ?LoopInterface $loop = null)
+    public function __construct(string $nameserver, ?LoopInterface $loop = null)
```

Comment:
This looks good, but I think you missed to apply the same changes for the `$nameserver` argument in the `TcpTransportExecutor`